### ISSUE TITLE
Statistics adjustment panel is not filterable on entity name

### ISSUE THREAD

**Issue by krazos** [home-assistant/frontend#14413]
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

When searching for an entity in the statistics adjustment panel in Developer Tools, the search function filters on `statistic_id` but does not filter on the friendly name of the entity, despite the relevant code in `src\panels\developer-tools\statistics\developer-tools-statistics.ts` indicating that the `Name` column should be filterable:

```
  private _columns = memoizeOne(
    (localize): DataTableColumnContainer => ({
      state: {
        title: "Name",
        sortable: true,
        filterable: true,
        grows: true,
        template: (entityState, data: any) =>
          html`${entityState
            ? computeStateName(entityState)
            : data.name || data.statistic_id}`,
      },
```

See example with screenshots below.

### Describe the behavior you expected

The statistics adjustable panels should be filterable on both `statistic_id` and friendly name for consistency with search functions elsewhere in the frontend. 

### Steps to reproduce the issue

1. Search for `solar_production`, a sub-string that appears within an `entity_id`:

![Screenshot from 2022-11-19 08-13-44](https://user-images.githubusercontent.com/18428022/202852720-aa5aac2b-0d07-4a5e-bc16-8587c0e41b6f.png)

2. Search for `solar production`, a sub-string that appears within a friendly name but not within the associated `entity_id`:

![Screenshot from 2022-11-19 08-14-21](https://user-images.githubusercontent.com/18428022/202852724-15e76932-3a5f-4465-ba96-66da0426e760.png)

### What version of Home Assistant Core has the issue?

core-2022.11.3

### In which browser are you experiencing the issue with?

Google Chrome 106.0.5249.119 and iOS App

**Comment by krazos** [comment]
Propose to keep this open as I do not believe this issue has been resolved.

**Comment by krazos** [comment]
Should this issue be marked closed as completed by #17098?
